pkg/hwinfo/hwinfo_windows.go

```diff
@@ -7,127 +7,226 @@ import (
 }
 
-type win32_BaseBoard struct {
-	Manufacturer *string
-	Product      *string
-	Model        *string
-	SerialNumber *string
+func listPCIDevices(errs *common.ErrorCollector) []*pciDeviceInfo {
+	query := windevice.DeviceQuery{
+		Enumerator: "PCI",
+		Flags:      deviceclass.Present,
+	}
+
+	result := make([]*pciDeviceInfo, 0)
+	var index int
+	err := query.Each(func(device windevice.Device) {
+		fullDisplayingName, err := device.Description()
+		if err != nil {
+			errs.Addf("could not get PCI device description %s", err.Error())
+			index++
+			return
+		}
+		friendlyName, _ := device.FriendlyName()
+
+		class, err := device.Class()
+		if err != nil {
+			errs.Addf("could not get PCI device class %s", err.Error())
+		}
+		location, err := device.LocationInformation()
+		if err != nil {
+			errs.Addf("could not get PCI device location %s", err.Error())
+		}
+		vendor, err := device.Manufacturer()
+		if err != nil {
+			errs.Addf("could not get PCI device vendor %s", err.Error())
+		}
+
+		description := class
+		if friendlyName != "" {
+			description = fmt.Sprintf("%s (%s)", friendlyName, class)
+		}
+
+		result = append(result, &pciDeviceInfo{
+			Address:     location,
+			DeviceType:  "",
+			VendorName:  vendor,
+			ProductName: fullDisplayingName,
+			Description: description,
+		})
+
+		index++
+	})
+
+	if err != nil {
+		errs.New(err)
+	}
+
+	return result
 }
 
-type win32_Processor struct {
-	Description               *string
-	Name                      *string
-	Manufacturer              *string
-	NumberOfCores             *uint32
-	NumberOfEnabledCore       *uint32
-	NumberOfLogicalProcessors *uint32
-
-	/*
-		Possible fields so we could use them later if we will need
-
-		DeviceID                  *string
-		SerialNumber              *string
-		SocketDesignation         *string
-		MaxClockSpeed             *uint32
-		ProcessorType             *uint16
-	*/
+func listUSBDevices(errs *common.ErrorCollector) []*usbDeviceInfo {
+	enumerator := "USB"
+
+	query := windevice.DeviceQuery{
+		Enumerator: enumerator,
+		Flags:      deviceclass.Present,
+	}
+
+	result := make([]*usbDeviceInfo, 0)
+	var index int
+	err := query.Each(func(device windevice.Device) {
+		fullDisplayingName, err := device.Description()
+		if err != nil {
+			errs.Addf("could not get USB device description %s", err.Error())
+			index++
+			return
+		}
+		deviceInstanceID, _ := device.DeviceInstanceID()
+		friendlyName, _ := device.FriendlyName()
+		location, _ := device.LocationInformation()
+		vendor, err := device.Manufacturer()
+		if err != nil {
+			errs.Addf("could not get USB device vendor %s", err.Error())
+		}
+		class, err := device.Class()
+		if err != nil {
+			errs.Addf("could not get USB device class %s", err.Error())
+		}
+		if class == "USB" {
+			class = ""
+		}
+
+		descriptionParts := []string{fullDisplayingName, friendlyName, class}
+		description := ""
+		for _, part := range descriptionParts {
+			if part != "" {
```

Comment:
Please use `len(x) == 0` and `len(x) > 0` instead of `x == ""` and `x != ""`.

Reply:
Not sure why? Both ways are idiomatic and used in the standard libraries. In both cases, the compiler output should be the same.

Reply:
Yes, but the first one is agnostic to the type. If the value going to change from `string` to `[]byte` the code will stay the same. I know that in this particular case it probably will never change, but anyways. These types are interchangeable, according to Go spec they mean the same:
```
A slice is a descriptor for a contiguous segment of an underlying array and provides access to a numbered sequence of elements from that array.

A string value is a (possibly empty) sequence of bytes. 
```

Check like `len(x) > 0 ` is easy to read, because it literally reads "length more than zero", the comparison `x != ""` looks like you are trying to find an exact value (of exact type). Moreover, sometimes the check needs to be adjusted, for example, skip not only empty strings, but short ones too. If you have `len(x) > 0` this is one digit to change: `len(x) > 2`, but in case of comparison `x != ""` you'll have to rewrite the whole line. Switching back will be painful too.

I'm okay if this change will be unaddressed in this PR, to avoid extra work, but is is something to consider for the future code and refactoring. ;)